exe-unit/components/gsb-http-proxy/tests/test_proxy.rs

```diff
@@ -0,0 +1,98 @@
+use actix_http::header::HeaderMap;
+use actix_web::{web, HttpResponse, Responder};
+use actix_web::{App, HttpServer};
+use chrono::Utc;
+use futures::StreamExt;
+use test_context::test_context;
+use ya_framework_basic::async_drop::DroppableTestContext;
+use ya_framework_basic::log::enable_logs;
+use ya_gsb_http_proxy::http_to_gsb::HttpToGsbProxy;
+use ya_gsb_http_proxy::message::GsbHttpCallMessage;
+use ya_gsb_http_proxy::response::GsbHttpCallResponseEvent;
+use ya_service_bus::typed as bus;
+
+#[test_context(DroppableTestContext)]
+#[serial_test::serial]
+pub async fn test_gsb_http_proxy(ctx: &mut DroppableTestContext) {
+    enable_logs(true);
+
+    start_proxy_http_server(ctx).await;
+    bind_proxy_to_gsb().await;
+    start_target_server(ctx).await;
+    bind_gsb_to_target().await;
+
+    let response = reqwest::get("http://127.0.0.1:8081/proxy")
+        .await
+        .expect("request should succeed");
+    let r: String = response.text().await.expect("response text expected");
+    assert_eq!(r, "correct");
+}
+
+async fn start_proxy_http_server(ctx: &mut DroppableTestContext) {
+    async fn proxy_endpoint() -> impl Responder {
+        let http_to_gsb = HttpToGsbProxy {
+            method: "GET".to_string(),
+            path: "target".to_string(),
+            body: None,
+            headers: HeaderMap::default(),
+        };
+
+        let mut stream = http_to_gsb
+            .pass(move |msg| bus::service(ya_gsb_http_proxy::BUS_ID).call_streaming(msg));
+        if let Ok(_r) = stream.next().await.unwrap() {
+            if let Ok(s) = String::from_utf8(_r.to_vec()) {
+                return s;
+            }
+        }
```

Comment:
I know that it was already implemented like this earlier, but I spent a lot of time trying to understand this test and I think the interface of this proxy is opposite of what user would expect:
I would expect that the parameters that I give to `HttpToGsbProxy` struct would be actually passed to `pass` function.

So from my perspective the interface could look like this:
- I create proxy object and pass target parameters like destination NodeId, address on destination and so on
    - Some of them might be optional some necessary, this depends
- I call `proxy.pass(...)` method and provide parameters of http which should be proxied.

Consumer of this proxy object shouldn't be forced to pass function that implements gsb call.

This can be for later

Reply:
I agree - once the component will take care of gsb binding this will make more sense.